Engine/source/core/util/tVector.h

```diff
@@ -63,11 +63,12 @@ extern bool VectorResize(U32 *aSize, U32 *aCount, void **arrayPtr, U32 newCount,
 template<class T>
 class Vector
 {
-  protected:
+public:
```

Comment:
quite dangerous, completely breaks encapsulation...

Reply:
Yes I forgot I made that change in there, wanted to bring it up and ask if there were another way to do it.. Maybe moving the Implement Struct to somewhere associated with Vector?

Reply:
An alternative solution is hard-coding the offsets:
```c++
IMPLEMENT_STRUCT( Vector< bool >,
   BoolVector,,
   "" )
   {"elementCount", "", 1, TYPE<U32>(), 0},
   {"arraySize", "", 1, TYPE<U32>(), 4},
   {"array", "", 1, TYPE<bool*>(), 8},
   //FIELD(mElementCount, elementCount, 1, "")
   //FIELD(mArraySize, arraySize, 1, "")
   //FIELD(mArray, elements, 1, "")
END_IMPLEMENT_STRUCT;
```

Reply:
Or the offsets could be computed and defined in tVector.h, but that sort of breaks encapsulation as well.

Alternatively, moving "IMPLEMENT_STRUCT( Vector<bool>" to tVector.h, but that breaks the current code-layout.